src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -99,7 +99,7 @@ public static string GetFullPath(string path, string basePath)
                     // No matching root, root to specified drive
                     // "D:Foo" and "C:\Bar" => "D:Foo"
                     // "D:\Foo" and "\\?\C:\Bar" => "\\?\D:\Foo"
-                    combinedPath = path.Insert(2, "\\");
+                    combinedPath = PathInternal.IsDevice(basePath) ? basePath.Substring(0,4) + path.Insert(2, "\\") : path.Insert(2, "\\");
```

Comment:
You could save some allocations by using CombineNoChecksInternal().  If you take a span and slice it on the original string you can end up with "C:" and "foo". Then you can slice basepath as well if you need the third one.

Reply:
Note that your other CombineNoChecks usages should probably be CombineNoChecksInternal. I don't think you want the rooting behavior.